src/client/app/containers/view.mode.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import * as fromRoot from '../reducers';
+import { UI, Snippet } from '../actions';
+import { environment, applyTheme } from '../helpers';
+import { Strings } from '../strings';
+
+@Component({
+    selector: 'view-mode',
+    template: `
+        <main [ngClass]="theme$|async">
+            <header class="command__bar">
+                <command class="title view-mode" [hidden]="isEmpty" icon="AppForOfficeLogo" [title]="snippet?.name"></command>
+            </header>
+            <editor [isViewMode]="true"></editor>
+            <footer class="command__bar command__bar--condensed">
+                <command id="feedback" [title]="Feedback" icon="Emoji2" (click)="feedback()"></command>
+                <command icon="Color" [title]="theme$|async" (click)="changeTheme()"></command>
+            </footer>
+        </main>
+    `
+})
+
+
+export class ViewMode implements OnInit, OnDestroy {
+    strings = Strings();
+    snippet: ISnippet;
+    showInfo: boolean;
+    paramsSub: Subscription;
+
+    constructor(
+        private _store: Store<fromRoot.State>,
+        private _route: ActivatedRoute
+    ) {
+        this._store.select(fromRoot.getCurrent).subscribe(snippet => {
+            this.snippet = snippet;
+        });
+    }
+
+    ngOnInit() {
+        this.paramsSub = this._route.params
+            .subscribe(async(params) => {
+                if (environment.current.host.toUpperCase() !== params.host.toUpperCase()) {
+                    environment.current.host = params.host.toUpperCase();
+                    // Update environment in cache
+                    environment.current = environment.current;
+                    await applyTheme(environment.current.host);
+                }
+
+                let urlSegments = this._route.snapshot.url;
+                switch (urlSegments[1].path) {
```

Comment:
Can you make sure the proper telemetry is in place for view URL, we need to be able to differentiate view mode vs editor import. We also need to know what they are viewing for public/private gist we can log the full id, for gist we need to log the hash value.